channels/api.py

```diff
@@ -414,6 +356,34 @@ def remove_user_role(channel_name, role, user):
     get_role_model(channel_name, role).group.user_set.remove(user)
 
 
+def get_post_type(*, text, url, article_content):
+    """
+    Returns the post type given the passed input values
+
+    Args:
+        text (str): the post text
+        url (str): the post url
+        article_content (dict): the post article text data
+    Returns:
+        str: the link type
+    """
+    if (
+        len(list(filter(lambda val: val is not None, [text, url, article_content])))
```

Comment:
Utility function would be nice here (`get_all_not_none` et al), but not important. You can take or leave that one

Reply:
Side note: this would probably be more fitting to implement in a [`validate`](https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation) method, though I realize this logic was already in `create` before this PR